src/applications/disability-benefits/all-claims/content/form0781/workflowChoicePage.jsx

```diff
@@ -193,3 +206,371 @@ export const mstAlert = () => {
+      questions online, we’ll remove PDF file you’ve uploaded.
+    </p>
+    <p>
+      <strong>Do you want to change your response to upload a PDF?</strong>
+    </p>
+  </>
+);
+
+const alertDescriptionSkip =
+  'We’ve removed information you’ve entered online about traumatic events';
+const alertDescriptionUpload =
+  'We’ve removed information about your traumatic events';
+const alertDescriptionOnline = 'We’ve removed your uploaded pdf';
+
+export const modalTitleSkip = 'Skip VA Form 21-0781?';
+export const modalTitleUpload = 'Change to upload a PDF?';
+export const modalTitleOnline = 'Change to answer questions online?';
+
+const deleteMentalHealthStatement = (data, setFormData) => {
+  const updatedData = { ...data };
+
+  mentalHealthKeys.forEach(key => {
+    delete updatedData[key];
+  });
+
+  setFormData(updatedData);
+};
+
+const deepCheck = value => {
+  switch (typeof value) {
+    case 'boolean':
+      return value === true;
+    case 'string':
+      return value.trim() !== '';
+    case 'number':
+      return true;
+    case 'object':
+      if (value === null) return false;
+
+      return Object.values(value).some(nestedValue => deepCheck(nestedValue));
+
+    default:
+      return false;
+  }
+};
+
+const checkMentalHealthData = formData => {
+  return mentalHealthKeys.some(key => {
+    if (!(key in formData)) {
+      return false;
+    }
+
+    return deepCheck(formData[key]);
+  });
+};
+
+const WorkflowChoicePage = props => {
+  const {
+    data,
+    goBack,
+    goForward,
+    setFormData,
+    contentBeforeButtons,
+    contentAfterButtons,
+  } = props;
+
+  const selectionField = 'view:mentalHealthWorkflowChoice';
+  const [previousWorkflowChoice, setPreviousWorkflowChoice] = useState(
+    data?.['view:previousMentalHealthWorkflowChoice'] ?? null,
+  );
+
+  const [hasError, setHasError] = useState(null);
+  const [showModal, setShowModal] = useState(false);
+  const [showAlert, setShowAlert] = useState(false);
+
+  useEffect(
+    () => {
+      if (showAlert) {
+        scrollTo('success-alert');
+      }
+    },
+    [showAlert],
+  );
+
+  const missingSelectionErrorMessage =
+    'A response is needed for this question.';
+
+  const missingSelection = (error, _fieldData, formData) => {
+    if (!formData?.[selectionField]) {
+      error.addError?.(missingSelectionErrorMessage);
+    }
+  };
+
+  const checkErrors = (formData = data) => {
+    const error = checkValidations(
+      [missingSelection],
+      data?.['view:mentalHealthWorkflowChoice'],
+      formData,
+    );
+
+    const result = error?.[0] || null;
+    setHasError(result);
+
+    return result;
+  };
+
+  const selectedChoice = data?.['view:mentalHealthWorkflowChoice'] ?? null;
+
+  const {
+    primaryText,
+    secondaryText,
+    modalContent,
+    alertContent,
+    modalTitle,
+  } = (() => {
+    switch (selectedChoice) {
+      case form0781WorkflowChoices.SUBMIT_PAPER_FORM:
```

Comment:
Ha this is clever, nice! I like how clear it makes the different modal states, love it